src/index.js

```diff
@@ -5,14 +5,43 @@ import {
     addRenderedClassNames, getRenderedClassNames
 } from './inject';
 
+// TODO(emily): Make a 'production' mode which doesn't prepend the class name
+// here, to make the generated CSS smaller.
+const makeClassName = (key, vals) => `${key}_${hashObject(vals)}`;
+
+// Find all of the references to descendant styles in a given definition,
+// generates a class name for each of them based on the class name of the
+// parent, and stores that name in a special `_names` object on the style.
+const findAndTagDescendants = (styles, base, names={}) => {
+    Object.keys(styles).forEach(key => {
+        if (key[0] === ':' || key[0] === '@') {
+            findAndTagDescendants(styles[key], base, names);
+        } else if (key[0] === '>' && key[1] === '>') {
+            findAndTagDescendants(styles[key], base, names);
+
+            const name = `${base}__${key.slice(2)}`;
+
+            names[key.slice(2)] = {
+                _isPlainClassName: true,
+                _className: name,
+            };
+
+            styles[key]._names = { [name]: true };
```

Comment:
That's how you wind up with more than one value in the array produced by `findNamesForDescendants`, but that doesn't yield multiple keys in the `_names` object, right?